File: PrimeFactorization.js
// reduces the parameter n into a product of only prime numbers
// and returns a list of those prime number factors
function primeFactorize(n) {
    var primeFactors = [];
    
    var primeFactor = 0;
        
    for (var i = 2; i <= n / i; ) {
        if (n % i == 0) {
            primeFactor = i;
            primeFactors.push(primeFactor);
            n /= i;
        } else {
            i++;
        }
    }
    
    if (primeFactor < n) {
        primeFactors.push(n);
    } else {
        primeFactors.push(primeFactor);
    }
        
    return primeFactors;
}

File: PrimeFactorization.py
# reduces the parameter n into a product of only prime numbers
# and returns a list of those prime number factors
def primeFactorize(n):
    primeFactors = []
    primeFactor = 0
    i = 2

    while i <= n / i:
        if n % i == 0:
            primeFactor = i
            primeFactors.append(i)
            n /= i
        else:
            i += 1

    if primeFactor < n:
        primeFactors.append(int(n))
    else:
        primeFactors.append(primeFactor)

    return primeFactors